icu4c/source/common/locid.cpp

```diff
@@ -256,6 +256,19 @@ U_NAMESPACE_BEGIN
 Locale::Nest::Nest() : language{'\0'}, script{'\0'}, region{'\0'}, variantBegin{0}, baseName{'\0'} {}
 Locale::Nest::~Nest() = default;
 
+Locale::Nest::Nest(std::string_view script, std::string_view region, uint8_t variantBegin)
+    : variantBegin(variantBegin), baseName{'\0'} {
+    // When moving from Heap to Nest the language field can be left untouched
+    // (as it has the same offset in both) and only the script and region fields
+    // need to be copied to their new locations, which is safe to do because the
+    // new locations come before the old locations in memory and don't overlap.
+    static_assert(sizeof Nest::language + sizeof Nest::script <= ULOC_LANG_CAPACITY);
+    static_assert(sizeof Nest::language + sizeof Nest::script + sizeof Nest::region <=
+                  ULOC_LANG_CAPACITY + ULOC_SCRIPT_CAPACITY);
```

Comment:
This would be clearer as
```
static_assert(offsetof(Nest, region) <= offsetof(Heap, script));
static_assert(offsetof(Nest, variantBegin) <= offsetof(Heap, region));
```
because one would see references to Heap without having to inspect how its fields are sized, and because it fits more closely to what the comment says: "the new locations come before the old locations in memory"